### ISSUE TITLE
The parking space count is wrong

### ISSUE THREAD

**Issue by cloutierjo** [sharedstreets/curb-map#11]
In the current viewer, the parking space count is wrong. When many rules apply at the same time for the same spot, this spot will be counted many times.

**Comment by cloutierjo** [comment]
I looked at the issue and here is the change that I propose:

1. Change the rule resolver (or actually make one that is isolated) that would output the active rules for each segment considering the time and later other criteria like vehicle types
2. update the counting algorithm to use that rule resolver

That resolver could then be used by the renderer as well to display only the applicable rules. I know that the "priority"  is going under change with the next version, and I would make sure that this logic can be reworked easily to the new category system.

I'm not going too deep in how the resolver would work since I find the actual way of doing it is good, just too tight with the rendering. The part where the rendering layer is used would be replaced with a filtering.

Does it seem a good start for you? I have some time to start working on this in the next weeks.

**Comment by cloutierjo** [comment]
This is still work in progress, but if you have any comment you’re welcome.

Two main issues remaining:
- The count is still linked to the rendering data and I couldn'T yet confirm if that count is now good 
- I don’t know yet how to fix the geometry when a partial segment need to be active, the render can still work OK because of the rendering priority and the parking count only use the shstLocationStart/End that the resolver update according to the rules which should also be good, but the geometry doesn't match the real segment which might cause trouble some other place.

**Comment by cloutierjo** [comment]
I've been able to confirm the count is now correct in a complex area where there are multiple regulation overlapping.
For the geometry, I believe it’s fine to leave it as is. I made sure the regulations are rendered in reverse order of priority, they are displayed correctly as it was before, but the regulations that were completely hidden are not rendered anymore which might be a small performance improvement, once the data is processed.

Let me know if any other improvement is required before we can merge this.
